project-koku/koku
# [COST-3892] Use aware timestamps for kafka message

## Jira Ticket

[COST-3892](https://issues.redhat.com/browse/COST-3892)

## Description

This change will split the time interval into a start and end time for the line items as this is an aware timestamp and the subs folks need a timezone identifier.

## Testing

1. Checkout Branch
2. set `ENABLE_SUBS_DEBUG=True`
3. Restart Koku with a subs worker `make docker-up-min-with-subs`
4. Create and load test customer data: `make create-test-customer` followed by `make load-test-customer-data test_source=AWS`
5. Check minio (localhost:9090) for a file with a similar path replacing the source with yours: `koku-bucket/org1234567/AWS/source=16a0f4fc-4acb-4e57-8194-5e9ee627c42b/date=2023-08-02/subs_d5ef5de6-79f4-4625-b1c9-562b13fc4fca_0.csv`
6. open the CSV and see that the `subs_start_time` and `subs_end_time` columns are suffixed with a `Z` to indicate UTC

Sample kafka message:
```
{"event_id": "06893602-d1f9-414d-8bf6-a90154e844be", "event_source": "cost-management", "event_type": "Snapshot", "account_number": "10001", "org_id": "1234567", "service_type": "RHEL System", "instance_id": "i-55555556", "timestamp": "2023-07-01T09:00:00Z", "expiration": "2023-07-01T10:00:00Z", "measurements": [{"value": "2", "uom": "vCPUs"}], "cloud_provider": "AWS", "hardware_type": "Cloud", "product_ids": ["479", "70"], "role": "Red Hat Enterprise Linux Workstation", "sla": "Self-Support", "usage": "Disaster Recovery", "billing_provider": "aws", "billing_account_id": "9999999999999"}
```
## Notes

...
